dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -222,6 +222,17 @@ def update(self, df_series):
 
         return self
 
+    def set_categorical_count_size(self, k: int):
+        """
+        Sets the number of top categories to be displayed when we call profile
+        and categorical_count in the statistics dictionary
+
+        :param k: the number we are changing top_k_categories to
+        :type k: int
+        :return: None
+        """
+        self._top_k_categories = k
+
```

Comment:
since the user won't have access to the function, lets use the `options` class to allow the set of this value.

e.g. https://github.com/capitalone/DataProfiler/blob/43c65ecad973d81bb6cc453b89bdaec2b46336c2/dataprofiler/profilers/profiler_options.py#L867
in TextProfilerOptions for Unstructured